test_aodndata/soop/test_soop_co2.py

```diff
@@ -0,0 +1,103 @@
+        # we expect this to succeed, so if the handler experiences an error, it is considered a
+        # "failed test"
+        handler = self.run_handler(GOOD_RT)
+
+        self.assertEqual(len(handler.file_collection), 1)
+        txt_file = handler.file_collection.filter_by_attribute_value('extension', '.txt')
+        nc_file = handler.file_collection.filter_by_attribute_value('extension', '.nc')
+
+        for t in txt_file:
+            self.assertEqual(t.publish_type, PipelineFilePublishType.ARCHIVE_ONLY)
+            self.assertEqual(t.archive_path,
+                             'IMOS/SOOP/SOOP-CO2/VLMJ_Investigator/REALTIME/2018/1/' + t.name)
+            self.assertTrue(t.is_archived)
+
+        for nc in nc_file:
+            self.assertEqual(nc.dest_path,
+                             'IMOS/SOOP/SOOP-CO2/VLMJ_Investigator/REALTIME/2018/1/' + nc.name)
+            self.assertTrue(nc.is_stored)
+
+    def test_frmap_file(self):
+        # test future Reef Map processing
+        handler = self.run_handler(GOOD_FRMAP)
+        self.assertEqual(len(handler.file_collection), 1)
+        f = handler.file_collection[0]
+        self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+        self.assertEqual(f.name, os.path.basename(GOOD_FRMAP))
+        self.assertEqual(f.dest_path,
+                         'Future_Reef_MAP/underway/RTM-Wakmatha/2015/WK201505N/FutureReefMap_GST_20150518T124011Z_9V2768_FV01.nc')
+        self.assertTrue(f.is_stored)
+
+    def test_good_nc_zip(self):
+
+        handler = self.handler_class(GOOD_ZIP,
+                                     check_params={'checks': ['cf']}
+                                     )
+        handler.run()
+        self.assertEqual(len(handler.file_collection), 3)
+
+        nc_files = handler.file_collection.filter_by_attribute_value('extension', '.nc')
+
+        for f in nc_files:
+            self.assertEqual(f.check_type, PipelineFileCheckType.NC_COMPLIANCE_CHECK)
+            self.assertEqual(f.publish_type, PipelineFilePublishType.HARVEST_UPLOAD)
+            self.assertEqual(f.name, os.path.basename(GOOD_NC))
+            self.assertEqual(f.dest_path,
+                             'IMOS/SOOP/SOOP-CO2/VNAA_Aurora-Australis/2017/AA1617_V3/' + f.name
+                             )
+
+        non_nc = handler.file_collection.filter_by_attribute_regex('extension', '[.](?!nc$)$')
+        for f in non_nc:
+            self.assertEqual(f.dest_path,
+                             'IMOS/SOOP/SOOP-CO2/VNAA_Aurora-Australis/2017/AA1617_V3/' + f.name)
+            self.assertEqual(f.check_type, PipelineFileCheckType.NONEMPTY_CHECK)
+            self.assertEqual(f.publish_type, PipelineFilePublishType.UPLOAD_ONLY)
+
+    def test_good_file_with_compliance_check(self):
+        # we also expect this to succeed, since the test file is known be CF compliant
+        self.run_handler(GOOD_NC, check_params={'checks': ['cf']})
+
+        if __name__ == '__main__':
```

Comment:
This is indented so that it is inside the test method, should be in the module root (i.e unindented).